clang/include/clang/CIR/Dialect/IR/CIRTypes.td

```diff
@@ -26,6 +26,28 @@ class CIR_Type<string name, string typeMnemonic, list<Trait> traits = []> :
   let mnemonic = typeMnemonic;
 }
 
+//===----------------------------------------------------------------------===//
+// IntType
+//===----------------------------------------------------------------------===//
+
+def CIR_IntType : CIR_Type<"Int", "int",
+    [DeclareTypeInterfaceMethods<DataLayoutTypeInterface>]> {
+  let summary = "Integer type with arbitrary precision up to a fixed limit";
+  let description = [{
+    CIR type that represents C/C++ primitive integer types.
+    Said types are: `char`, `short`, `int`, `long`, `long long`, and their \
```

Comment:
Should we add a type aliases for char? Example `!cir.uchar` -> `!cir.int<u, 8>` and `!cir.schar` -> `!cir.int<s, 8>`. Doesn't need to be part of this PR, just bringing up the subject.

Reply:
I like the idea.
Actually, I think ClangIR would be quite more readable if we added aliases for all types (`int`, `short`, etc.).
My only concern with aliases is losing width information since some of then can have different widths on different systems, but, as far as I know, `char`s are guaranteed to always be a single byte right?

Reply:
> Actually, I think ClangIR would be quite more readable if we added aliases for all types (`int`, `short`, etc.).

That's an interesting point!

> My only concern with aliases is losing width information

I wouldn't be too worried here because the alias definition in the top of the file will contain the map for the right size - OTOH I believe it brings more insights (for compiler devs) to have the size in the face instead of proxied by the alias (e.g. `!u32` versus `!unsigned`).

> chars are guaranteed to always be a single byte right?

Yes, but bytes are not guaranteed to be 8 bits LOL - though this isn't too much relevant for us and we are free to make safely make this assumption nowadays